packages/xo-web/src/common/xo/index.js

```diff
@@ -2385,15 +2385,26 @@ export const editResourceSet = (
   })::tap(subscribeResourceSets.forceRefresh)
 
 export const deleteResourceSet = async id => {
+  const vms = await getAllObjects({ resourceSet: resolveId(id) })
```

Comment:
Better do this logic in the server side, the first reason is that the API method can be called from other API clients for example `xo-cli`.